src/coreclr/gc/gcinterface.dac.h

```diff
@@ -46,9 +46,15 @@ class dac_heap_segment {
 // of a generation and its allocation context.
 class dac_generation {
 public:
-    gc_alloc_context allocation_context;
-    DPTR(dac_heap_segment) start_segment;
-    uint8_t* allocation_start;
+#define ALL_FIELDS
+#define DEFINE_FIELD(field_name, field_type) field_type field_name;
```

Comment:
is there a case where `ALL_FIELDS` would not be defined?

Reply:
The only case where `ALL_FIELDS` is not defined is `gc.cpp` where we generate the offset array. There, we need to make sure we do not attempt to use the `offsetof` operator and emit a -1 directly.

Reply:
I see; makes sense